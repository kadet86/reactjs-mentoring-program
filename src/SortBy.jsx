// @flow
import * as React from 'react';
import styled from 'styled-components';

type Props = {
    sortBy: string,
    onSortByChange: (event: SyntheticEvent<HTMLElement>) => void,
};

const SortByContainer = styled.div`
    float: right;

    .sort-by__link {
        padding-right: 10px;
    }

    .sort-by__link--active {
        color: red;
    }
    
    label {
        padding-right: 10px;
    }
`;

export default function SortBy({ sortBy, onSortByChange }: Props) {
    return (
        <SortByContainer>
            <label>Sort by</label>
            <a
                className={
                    'sort-by__link' +
                    (sortBy === 'release_date' ? ' sort-by__link--active' : '')
                }
                data-value="release_date"
                href="javascript:void(0);"
                onClick={onSortByChange}
            >
                release date
            </a>
            <a
                className={
                    'sort-by__link' +
                    (sortBy === 'vote_average' ? ' sort-by__link--active' : '')
                }
                data-value="vote_average"
                href="javascript:void(0);"
                onClick={onSortByChange}
            >
                rating
            </a>
        </SortByContainer>
    );
}
