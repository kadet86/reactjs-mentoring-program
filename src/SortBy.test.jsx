import React from 'react';
import {shallow} from 'enzyme';
import SortBy from './SortBy';

describe('SortBy component', () => {
    it('has persistent default snapshot', () => {
        const component = shallow(<SortBy />);
        expect(component.html()).toMatchSnapshot();
    });

    it('has persistent snapshot with sortBy="release_date"', () => {
        const component = shallow(<SortBy sortBy="release_date"/>);
        expect(component.html()).toMatchSnapshot();
    });

    it('has persistent snapshot with sortBy="vote_average"', () => {
        const component = shallow(<SortBy sortBy="vote_average"/>);
        expect(component.html()).toMatchSnapshot();
    });
});
