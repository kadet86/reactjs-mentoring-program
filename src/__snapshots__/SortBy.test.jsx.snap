// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy component has persistent default snapshot 1`] = `"<div class=\\"sort-by\\"><label>Sort by</label><a class=\\"sort-by__link\\" data-value=\\"release_date\\" href=\\"javascript:void(0);\\">release date</a><a class=\\"sort-by__link\\" data-value=\\"vote_average\\" href=\\"javascript:void(0);\\">rating</a></div>"`;

exports[`SortBy component has persistent snapshot with sortBy="release_date" 1`] = `"<div class=\\"sort-by\\"><label>Sort by</label><a class=\\"sort-by__link sort-by__link--active\\" data-value=\\"release_date\\" href=\\"javascript:void(0);\\">release date</a><a class=\\"sort-by__link\\" data-value=\\"vote_average\\" href=\\"javascript:void(0);\\">rating</a></div>"`;

exports[`SortBy component has persistent snapshot with sortBy="vote_average" 1`] = `"<div class=\\"sort-by\\"><label>Sort by</label><a class=\\"sort-by__link\\" data-value=\\"release_date\\" href=\\"javascript:void(0);\\">release date</a><a class=\\"sort-by__link sort-by__link--active\\" data-value=\\"vote_average\\" href=\\"javascript:void(0);\\">rating</a></div>"`;
